api/src/main/java/io/opentelemetry/trace/Tracestate.java

```diff
@@ -235,28 +235,35 @@ public static Entry create(String key, String value) {
 
   // Key is opaque string up to 256 characters printable. It MUST begin with a lowercase letter, and
   // can only contain lowercase letters a-z, digits 0-9, underscores _, dashes -, asterisks *, and
-  // forward slashes /.
+  // forward slashes /.  For multi-tenant vendor scenarios, an at sign (@) can be used to prefix the
+  // vendor name.
   private static boolean validateKey(String key) {
-    if (key.length() > KEY_MAX_SIZE
-        || key.isEmpty()
-        || key.charAt(0) < 'a'
-        || key.charAt(0) > 'z') {
+    if (key.length() > KEY_MAX_SIZE || key.isEmpty() || !isNumberOrDigit(key.charAt(0))) {
       return false;
     }
+    int atSeenCount = 0;
     for (int i = 1; i < key.length(); i++) {
       char c = key.charAt(i);
-      if (!(c >= 'a' && c <= 'z')
+      if (!isNumberOrDigit(c)
           && !(c >= '0' && c <= '9')
```

Comment:
I think this condition is covered by the previous one.

Reply:
I think so too.  Also, `isNumberOrDigit` is always negated (per IDEA warning). I'll fix both of these issues and update the PR for @breedx-nr

Reply:
fixed and pushed